spatial_cells/block_adjust_gpu.cpp

```diff
@@ -754,4 +614,266 @@ void adjust_velocity_blocks_in_cells(
    }
 }
 
+void clear_maps_caller(const uint nCells,
+                       const size_t largestSizePower,
+                       gpuStream_t stream,
+                       const size_t offset
+   ) {
+   const size_t largestMapSize = std::pow(2,largestSizePower);
+   // fast ceil for positive ints
+   //const size_t blocksNeeded = 1 + ((largestMapSize - 1) / Hashinator::defaults::MAX_BLOCKSIZE);
+   size_t blocksNeeded = 1 + floor(sqrt(largestMapSize / Hashinator::defaults::MAX_BLOCKSIZE)-1);
+   blocksNeeded = std::max((size_t)1, blocksNeeded);
+   dim3 grid1(blocksNeeded,2*nCells,1);
+   batch_reset_all_to_empty<<<grid1, Hashinator::defaults::MAX_BLOCKSIZE, 0, stream>>>(
+      dev_allMaps+2*offset
+      );
+   CHK_ERR( gpuPeekAtLastError() );
+   CHK_ERR( gpuStreamSynchronize(stream) );
+}
+
+
+void batch_adjust_blocks_caller(
+   dccrg::Dccrg<spatial_cell::SpatialCell,dccrg::Cartesian_Geometry>& mpiGrid,
+   const vector<CellID>& cellsToAdjust,
+   const uint cellOffset,
+   uint &out_largestBlocksToChange,
+   uint &out_largestBlocksBeforeOrAfter,
+   const uint popID
+   ) {
+
+   const uint nCells = cellsToAdjust.size();
+   if (nCells == 0) {
+      return;
+   }
+   const gpuStream_t baseStream = gpu_getStream();
+
+   // Resizes are faster this way with larger grid and single thread
```

Comment:
I trust you tested, but it's probably going to depend massively on the system simulated and be very fickle. And there'll be other kernels. But should there be some interface to allow the user to set these at runtime without recompiling? Say to go through parameters (semi-)automatically and determine useful grid and block sizes for a given setup? One thing I noticed in a recent GPU course I took was to not think of these parameters too much as a function of actual problem size/cell count etc but in a more generic way...